go/common/metrics/metrics.go

```diff
@@ -0,0 +1,50 @@
+package metrics
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/metrics/exp"
+
+	gethlog "github.com/ethereum/go-ethereum/log"
+	gethmetrics "github.com/ethereum/go-ethereum/metrics"
+)
+
+// Service provides the metrics for the host
+// it registers the gethmetrics Registry
+// and handles the metrics server
+type Service struct {
+	registry gethmetrics.Registry
+	port     uint
+	logger   gethlog.Logger
+}
+
+func New(enabled bool, port uint, logger gethlog.Logger) *Service {
+	gethmetrics.Enabled = enabled
+	return &Service{
+		registry: gethmetrics.NewRegistry(),
+		port:     port,
+		logger:   logger,
+	}
+}
+
+// Start starts the metrics server
+func (m *Service) Start() {
+	// metrics not enabled
+	if !gethmetrics.Enabled {
+		return
+	}
+	// starts the metric server
+	address := fmt.Sprintf("%s:%d", "0.0.0.0", m.port)
+	m.logger.Info("HTTP Metric server started at %s", address)
+	// TODO re-write this http server so to have a stop method
+	exp.Setup(address)
```

Comment:
I noticed geth also does this on setup:

```
	// Start system runtime metrics collection
	go metrics.CollectProcessMetrics(3 * time.Second)
```

let's do that as well.

Reply:
I had considered enabling that but when I dug down I saw that method uses the default registry under the hood. 

Adding a todo and I'll create a new PR with the process metrics.